src/Storefront/Controller/AddressController.php

```diff
@@ -140,9 +140,9 @@ public function checkoutSwitchDefaultAddress(RequestDataBag $data, SalesChannelC
 
         $this->addFlash(self::SUCCESS, $this->trans('account.addressDefaultChanged'));
 
-        return new RedirectResponse(
-            $this->generateUrl('frontend.account.addressmanager.get')
-        );
+        $request->request->set('redirectTo', $data->get('redirectTo', 'frontend.account.addressmanager.get'));
```

Comment:
You can use the Request here directly and drop the RequestDataBag alltogether, as you did in `addressManagerSwitch`